### PR TITLE
Switch default HBHENoiseFilter settings to Run2-25ns configuration V2 - 76X

### PR DESCRIPTION
**Pull request by dertexaner**
This is take-2 to switch the default HBHENoiseFilterResultProducer settings to Run2-25ns configuration such that users get correct noise flags out of the box for the bulk of 2015 data (and beyond) which is with 25ns bunch spacing.
As per @slava77's request (see PR #11778 ), Eras is replaced by BunchSpacingProducer in HBHENoiseFilterResultProducer.
In the default settings, BunchSpacingProducer is enabled, which should automatically switch between Run2 50ns and 25ns settings. However, users will have to manually configure and rerun HBHENoiseFilterResultProducer in case of any Run1 50ns data analysis.
@abdoulline @igv4321 @mariadalfonso

### CODE REVIEW

File: CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py [hunk 1]
```diff
@@ -6,13 +6,10 @@
...
+    defaultDecision = cms.string("HBHENoiseFilterResultRun2Loose"),
     minNumIsolatedNoiseChannels = cms.int32(10),
     minIsolatedNoiseSumE = cms.double(50.0),
-    minIsolatedNoiseSumEt = cms.double(25.0)
+    minIsolatedNoiseSumEt = cms.double(25.0),
+    useBunchSpacingProducer = cms.bool(True)
 )
-
-from Configuration.StandardSequences.Eras import eras
-eras.run2_common.toModify(HBHENoiseFilterResultProducer, IgnoreTS4TS5ifJetInLowBVRegion=False)
```
Comment (slava77) [inline CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py]: I think this (IgnoreTS4TS5ifJetInLowBVRegion) is still useful and should stay.
My request was towards removing configuration-level settings to switch between 50ns or 25 ns in the same major period of data taking as we had in 2015

run1 vs run2 configuration changes should be OK for some time (trying to keep this down to detector level changes).

**Comment by dertexaner** [comment]
My impression was that the dependence on Eras (I guess, more specifically on run2_25ns_specific vs run2_50ns_specific) was proving to be problematic and resulted in effectively everyone rerunning the HBHENoiseFilterResultProducer at the analysis level. The current PR would eliminate this, with the very limited exception of Run1 analyses. 
Is there any risk for "eras.run2_common" not to be specified properly?  If so, I'd prefer keeping Eras out of the picture altogether since it only caused confusion so far.

**Comment by slava77** [comment]
On 10/14/15 5:08 PM, dertexaner wrote:
> My impression was that the dependence on Eras (I guess, more
> specifically on run2_25ns_specific vs run2_50ns_specific) was proving to
> be problematic and resulted in effectively everyone rerunning the
> HBHENoiseFilterResultProducer at the analysis level. The current PR
> would eliminate this, with the very limited exception of Run1 analyses.
> Is there any risk for "eras.run2_common" not to be specified properly?
> If so, I'd prefer keeping Eras out of the picture altogether since it
> only caused confusion so far.
> 

The problematic part is having to configure 25ns and 50ns separately
during run2,
-  some datasets can have both kinds of fills data included
- people processing both 25ns and 50ns datasets (MC and data) will have less
	changes to take care of and will not have to remember to make the switch
	if we manage to eliminate this configuration level dependence of 25ns
vs 50ns.

**Comment by dertexaner** [comment]
I think I understand . But just to clarify, the current configuration requires no change in 50ns Run2 and 25ns Run2 scenarios, and it doesn't depend on any Eras customization. 
If your point is that Run1 vs Run2 changes should also be automated, then yes, we will need to rely on Eras. Is this the case?

**Comment by slava77** [comment]
On 10/14/15 6:04 PM, dertexaner wrote:
> I think I understand . But just to clarify, the current configuration
> requires no change in 50ns Run2 and 25ns Run2 scenarios, and it doesn't
> depend on any Eras customization.

Your new code changes indeed make it automatic at runtime to figure out
if it's 50ns or 25 ns.
Meaning that the same cmsRun config.py will work if input file has 50ns
or 25 ns data.

Previously, even with eras and 25ns specific settings you would have to
have different config.py files
to correctly process inputs with 25ns, to be running separately on a
different config py file for inputs with 50 ns. (I thought it was kind
of obvious)

> If your point is that Run1 vs Run2 changes should also be automated,
> then yes, we will need to rely on Eras. Is this the case?

considering use cases (rarely run1 and run2 data mixed together in one
analysis and the same
release), using different configurations (via eras, preferably, now) is OK.

File: CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py (hunk 1 above)
Comment (slava77) [inline CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py]: I'm still expecting the 
``` eras.run2_common.toModify(HBHENoiseFilterResultProducer, IgnoreTS4TS5ifJetInLowBVRegion=False) ``` 
to return back to the file and "True" to be in the run1 config.

**Comment by slava77** [comment]
also, please cleanup customise_DQM_25ns function so that it no longer has the modification of HBHENoiseFilterResultProducer
https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_0_pre7/SLHCUpgradeSimulations/Configuration/python/postLS1Customs.py#L193

Thank you.

File: CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py (hunk 1 above)
Comment (dertexaner) [inline CommonTools/RecoAlgos/python/HBHENoiseFilterResultProducer_cfi.py]: Ready.

**Comment by slava77** [comment]
@dertexaner 
please clarify if you can update the customise_DQM_25ns  mentioned in https://github.com/cms-sw/cmssw/pull/11789#issuecomment-148420245
With current changes in this PR it already becomes redundant (changes in the PSet there will do nothing)

**Comment by dertexaner** [comment]
Hopefully that worked.

**Comment by slava77** [comment]
@cmsbuild please test

@dertexaner thank you

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/11789	 9acf9a8
- technical change to automate choice of default outputs depending on the bunch spacing
- jenkins tests pass and comparisons with baseline shows no differences
- additionally tested locally in CMSSW_7_6_X_2015-10-19-1100, including run2 134.{7,8}0* workflows: no changes were observed and comparison of expanded run1 and run2 reco+pat workflow steps shows expected changes

**Comment by davidlange6** [comment]
+1

**Pull request merged**